core/src/com/group66/game/helpers/LevelLoader.java

```diff
@@ -89,14 +89,15 @@ public static void loadLevel(BallManager ballManager1, BallManager ballManager2)
             e.printStackTrace();
         }
     }
-
+    
     /**
-     * Generate a random level.
+     * Generate a random level for one player.
      */
     public static void generateLevel(BallManager ballManager) {
         Random rand = new Random();
-        //int numRows = 2 + r.nextInt(5); // 2-6 rows
-        int numRows = 4;
+        DifficultyManager difficultyManager = new DifficultyManager();
```

Comment:
I ment more as in like the dificultyManager is initialized somewhere like in bustamove or mainmenu screen and passed to the settings screen and level loader. So it would also have a local version of its difficulty, as now the actual  difficulty is still static and has a value for all the created classes.

But as we are kinda hitting a deadline, this is Ok for now